Add CrUX Documentation (#3078)
GoogleChrome/developer.chrome.com: site/en/docs/crux/methodology/index.md

# Optional
# How to add a new author
# /docs/handbook/how-to/add-an-author/
authors:
  - simonhearne

# Optional
# How to a new tag
# /docs/handbook/how-to/add-a-tag/
tags:
  - performance
---

## Eligibility

At the core of the CrUX dataset are individual user experiences, which are aggregated into page-level and origin-level distributions. This section documents user eligibility and the requirements for pages and origins to be included in the dataset. All eligibility criteria must be satisfied in order for an experience to be included in page-level data available in PageSpeed Insights and the CrUX API: [User](#user-eligibility), [Origin](#origin-eligibility) and [Page](#page-eligibility). Experiences which meet the User and Origin criteria (but not Page) will be included in the origin-level data available in all CrUX data sources.

Pages and origins will be automatically included or removed from the dataset if their eligibility changes over time. There is not currently a way to manually submit pages or origins for inclusion.

### User {: #user-eligibility}

For a **user** to have their experiences aggregated in the CrUX dataset, they must meet the following criteria:

> 1. Enable [usage statistic reporting](https://support.google.com/chrome/answer/96817)
> 2. Sync their [browser history](https://support.google.com/chrome/answer/185277)
> 3. Not have a [Sync passphrase](https://support.google.com/chrome/answer/165139?co=GENIE.Platform%3DAndroid#zippy=%2Ccreate-a-passphrase) set
> 4. Use a supported platform

Reviewer: ```suggestion
1. Enable [usage statistic reporting](https://support.google.com/chrome/answer/96817).
2. Sync their [browser history](https://support.google.com/chrome/answer/185277).
3. Not have a [Sync passphrase](https://support.google.com/chrome/answer/165139?co=GENIE.Platform%3DAndroid#zippy=%2Ccreate-a-passphrase) set.
4. Use a supported platform.
```

The current supported platforms are:

> - Desktop versions of Chrome including Windows, MacOS, ChromeOS, and Linux operating systems
> - Android versions of Chrome, including native apps using [Custom Tabs](/docs/android/custom-tabs/) and [WebAPKs](https://web.dev/webapks/)

Reviewer: ```suggestion
- Desktop versions of Chrome including Windows, MacOS, ChromeOS, and Linux operating systems.
- Android versions of Chrome, including native apps using [Custom Tabs](/docs/android/custom-tabs/) and [WebAPKs](https://web.dev/webapks/).
```

There are a few notable exceptions that do not provide data to the CrUX dataset:

> - Chrome on iOS
> - Native Android apps using WebView
> - Other Chromium browsers (e.g. [Microsoft Edge](https://www.microsoft.com/edge))

Reviewer: ```suggestion
- Chrome on iOS.
- Native Android apps using WebView.
- Other Chromium browsers (for example [Microsoft Edge](https://www.microsoft.com/edge)).
```

Chrome does not publish data about the proportions of users that meet these criteria. You can learn more about the data we collect in the [Chrome Privacy Whitepaper](https://www.google.com/chrome/privacy/whitepaper.html#usagestats).

### Origin {: #origin-eligibility }

An [**origin**](https://developer.mozilla.org/docs/Glossary/Origin) represents an entire website, addressable by a URL like `https://www.example.com`. For an origin to be included in the CrUX dataset it must meet two requirements:

1. Publicly discoverable
2. Sufficiently popular

An origin is considered to be publicly discoverable if its **root page** is discoverable. For example, the root page of the origin `https://www.example.com` would have a URL of `https://www.example.com/`. If the root page has any HTTP redirects, the origin will be assessed based on the redirected page as experienced by real users.

Any page will **not** meet the discoverability requirement if **any** of the following conditions are met, including root pages for the origin dataset:

> - The page is served with an HTTP [status code](https://developer.mozilla.org/docs/Web/HTTP/Status) other than `200`
> - The page is served with an HTTP `X-Robots-Tag: noindex` [header](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag#xrobotstag-implementation)
> - The document includes a `<meta name="robots" content="noindex">` [meta tag](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag)
> - The URL is disallowed by [robots.txt](https://developers.google.com/search/docs/advanced/robots/intro)

Reviewer: ```suggestion
- The page is served with an HTTP [status code](https://developer.mozilla.org/docs/Web/HTTP/Status) other than `200`
- The page is served with an HTTP `X-Robots-Tag: noindex` [header](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag#xrobotstag-implementation)
- The document includes a `<meta name="robots" content="noindex">` [meta tag](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag)
- The URL is disallowed by [robots.txt](https://developers.google.com/search/docs/advanced/robots/intro)
```
```suggestion
- The page is served with an HTTP [status code](https://developer.mozilla.org/docs/Web/HTTP/Status) other than `200`.
- The page is served with an HTTP `X-Robots-Tag: noindex` [header](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag#xrobotstag-implementation).
- The document includes a `<meta name="robots" content="noindex">` [meta tag](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag).
- The URL is disallowed by [robots.txt](https://developers.google.com/search/docs/advanced/robots/intro).
```

### CrUX Dashboard {: #tool-crux-dash}

The [CrUX Dashboard](https://web.dev/chrome-ux-report-data-studio-dashboard/) is a Data Studio dashboard that allows you to query and render CrUX data into an interactive dashboard, as well as exporting PDF reports.

The dashboard provides visualization of all CrUX metrics in monthly trends, with data available back to 2017. Data can be split by form factor to compare mobile / tablet / desktop performance and performance goals are available to create red-amber-green visualizations.

The CrUX dashboard does not support effective connection type or country dimensions, so all global data is presented in the reports. Page-level data is not available and percentile values are calculated from coarse histogram data so are approximate.

### CrUX API {: #tool-crux-api}

The [CrUX API](https://web.dev/chrome-ux-report-api/) provides programmatic access to CrUX data by page or origin.

The API provides web vitals metrics both by origin and at page-level and the data is updated daily. The only values provided for metrics are calculated from the previous 28 days as a rolling window, no historical data is available via the API.

> [Read more in the API documentation](/crux/docs/api/).

Reviewer 2: The distinction between the web.dev / DCC pages should be: "how to use it" on web.dev, and "official product docs" on DCC. I'm open to whatever works best for consistency and clarity. Agreed that updating the first link so that it goes to the product page or documentation page makes more sense. We could still link to the web.dev guides while clarifying that this is where readers can learn more about these tools.

PageSpeed Insights does not provide historical data, and does not include country or effective connection type dimensions.

### PageSpeed Insights API {: #tool-psi-api}

The [PageSpeed Insights API](https://developers.google.com/speed/docs/insights/v5/get-started) offers programmatic access to the data shown in PageSpeed Insights, including Core Web Vitals data from CrUX.

This API integrates well into existing SEO tooling and workflows, allowing CrUX data to be included in automated reports and analyses.

As in the web version, the PageSpeed Insights API has no historical data and is limited to the core web vitals. Country and effective connection type dimensions are not included.

### Search Console {: #tool-gsc}

[Search Console](https://search.google.com/search-console) shows how CrUX data influences the [Page Experience](https://developers.google.com/search/docs/advanced/experience/page-experience) ranking factor by URL and URL group.

> Search console presents data at the page-level, with individual pages grouped into URL groups with aggregate core web vitals values. This provides a quick indication of which sections of a site are potentially impacting the page experience ranking factor.

Reviewer 2: ```suggestion
Search Console presents data at the page-level, with individual pages grouped into URL groups with aggregate Core Web Vitals values. This provides a quick indication of which sections of a site are potentially impacting the page experience ranking factor.
```

Country-level datasets are provided in addition to the global dataset, with the standard eligibility requirements applied at a country level. A table is [provided for each country](../bigquery/#schema-country-summary), as well as [summary tables](../bigquery/#schema-raw-tables) which include the country code as a column.

### Optional dimensions {: #optional-dimensions}

As of the [May 2022](../release-notes/#202205) release, the CrUX dataset supports optional dimensions. Previously, a form factor and effective connection type (ECT) combination must have independently met the _sufficiently popular_ criterion or else it would be excluded from the page or origin record. With this feature, experiences on different ECTs could be combined by their common form factor, and the corresponding ECT value will be `NULL`. Experiences on different form factors may also be combined, and both the ECT and form factor values will be `NULL`.

> Previously, form factor and effective connection type were required columns in our BigQuery tables. This meant that when we did not have sufficient coverage to express the histogram densities in the specific rows (e.g., form factor = phone, effective connection type = 2G), we were dropping the entire origin from the dataset. With optional dimensions, we made the form factor and effective connection type optional (NULLABLE) and therefore, we're now able to publish overall histogram densities in such cases; that is, we may set the effective connection type value to NULL indicating "all effective connection types", or we may set both effective connection type and form factor to NULL indicating "all effective connection types" and "all form factors".

Author: I think I prefer the more verbose version, @rviscomi thoughts?

> Previously, form factor and effective connection type were required columns in our BigQuery tables. This meant that when we did not have sufficient coverage to express the histogram densities in the specific rows (e.g., form factor = phone, effective connection type = 2G), we were dropping the entire origin from the dataset. With optional dimensions, we made the form factor and effective connection type optional (NULLABLE) and therefore, we're now able to publish overall histogram densities in such cases; that is, we may set the effective connection type value to NULL indicating "all effective connection types", or we may set both effective connection type and form factor to NULL indicating "all effective connection types" and "all form factors".

Reviewer 2: Verbose version SGTM if that's what you think is clearer for readers.

@tunetheweb we're only referring to the public dataset here.

> Previously, form factor and effective connection type were required columns in our BigQuery tables. This meant that when we did not have sufficient coverage to express the histogram densities in the specific rows (e.g., form factor = phone, effective connection type = 2G), we were dropping the entire origin from the dataset. With optional dimensions, we made the form factor and effective connection type optional (NULLABLE) and therefore, we're now able to publish overall histogram densities in such cases; that is, we may set the effective connection type value to NULL indicating "all effective connection types", or we may set both effective connection type and form factor to NULL indicating "all effective connection types" and "all form factors".

Reviewer 3: Yeah I just think saying "Previously, form factor and effective connection type were required columns in our BigQuery tables" is confusing as most people only know the public dataset that is updated once a month.

This change also affects the daily API, and yes the reason that that is affected too, is that they are required columns in our private BigQuery dataset. But that's the point that I think is confusing.

So I'm suggesting dropping any mention of "required columns" or "BigQuery tables" to avoid that confusion.

> Previously, form factor and effective connection type were required columns in our BigQuery tables. This meant that when we did not have sufficient coverage to express the histogram densities in the specific rows (e.g., form factor = phone, effective connection type = 2G), we were dropping the entire origin from the dataset. With optional dimensions, we made the form factor and effective connection type optional (NULLABLE) and therefore, we're now able to publish overall histogram densities in such cases; that is, we may set the effective connection type value to NULL indicating "all effective connection types", or we may set both effective connection type and form factor to NULL indicating "all effective connection types" and "all form factors".

Reviewer 2: Synced with Simon on this. It's ok to leave as-is.

---
# Required
layout: 'layouts/doc-post.njk'

# Required

> title: CrUX Methodology

Reviewer: yes please :)

# Required
date: 2022-06-23

# Optional
# Include an updated date when you update your post
# updated: 2020-10-16

# Optional
# How to add a new author
# /docs/handbook/how-to/add-an-author/
#authors:

# Optional
# How to a new tag
# /docs/handbook/how-to/add-a-tag/
tags:
  - web-vitals
---

## Eligibility

At the core of the CrUX dataset are individual user experiences, which are aggregated into page-level and origin-level distributions. This section documents user eligibility and the requirements for pages and origins to be included in the dataset. All eligibility criteria must be satisfied in order for an experience to be included in page-level data available in PageSpeed Insights and the CrUX API: [User](#user-eligibility), [Origin](#origin-eligibility) and [Page](#page-eligibility). Experiences which meet the User and Origin criteria (but not Page) will be included in the origin-level data available in all CrUX data sources.

Pages and origins will be automatically included or removed from the dataset if their eligibility changes over time. There is not currently a way to manually submit pages or origins for inclusion.

### User {: #user-eligibility}

> For a **user** to have their experiences aggregated in the CrUX dataset, they must meet the following criteria:

Reviewer 4: I'm not native english speaker, is it right to say 'they must' for a user?

Reviewer 2: This sounds ok to me

4. Use a supported platform.

The current supported platforms are:

- Desktop versions of Chrome including Windows, MacOS, ChromeOS, and Linux operating systems.
- Android versions of Chrome, including native apps using [Custom Tabs](/docs/android/custom-tabs/) and [WebAPKs](https://web.dev/webapks/).

There are a few notable exceptions that do not provide data to the CrUX dataset:

- Chrome on iOS.
- Native Android apps using WebView.
- Other Chromium browsers (for example [Microsoft Edge](https://www.microsoft.com/edge)).

Chrome does not publish data about the proportions of users that meet these criteria. You can learn more about the data we collect in the [Chrome Privacy Whitepaper](https://www.google.com/chrome/privacy/whitepaper.html#usagestats).

### Origin {: #origin-eligibility }

An [**origin**](https://developer.mozilla.org/docs/Glossary/Origin) represents an entire website, addressable by a URL like `https://www.example.com`. For an origin to be included in the CrUX dataset it must meet two requirements:

1. Publicly discoverable
2. Sufficiently popular

> An origin is considered to be publicly discoverable if its **root page** is discoverable. For example, the root page of the origin `https://www.example.com` would have a URL of `https://www.example.com/`. If the root page has any HTTP redirects, the origin will be assessed based on the redirected page as experienced by real users.

Reviewer 2: The list below goes into these specifics

Any page will **not** meet the discoverability requirement if **any** of the following conditions are met, including root pages for the origin dataset:

> - The page is served with an HTTP [status code](https://developer.mozilla.org/docs/Web/HTTP/Status) other than `200`.

Reviewer 2: A page that redirects may still be discoverable, but for attribution purposes the starting URL/origin wouldn't matter anymore, only the final one.

> - The page is served with an HTTP [status code](https://developer.mozilla.org/docs/Web/HTTP/Status) other than `200`.

Author: How about updating that bullet to  
> ...other than `200` (after redirects)
?

- The page is served with an HTTP `X-Robots-Tag: noindex` [header](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag#xrobotstag-implementation).

> - The document includes a `<meta name="robots" content="noindex">` [meta tag](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag).

Reviewer 4: Is this new? I remember that John Mueller stated that even noindex pages could be added..

Reviewer 2: Not sure what John was referring to. An origin is not public if its root page is not discoverable. If an origin is public, it may contain aggregated data from all underlying pages, public _or_ private. We go into more detail on this a couple of paragraphs down.

A website's architecture may complicate how its data is represented in CrUX. For example, single page apps (SPAs) may use a JavaScript-based _route transition_ scheme to move between pages, as opposed to traditional anchor-based page navigations. These transitions appear as new page views to the user, but to Chrome and the underlying platform APIs the entire experience is attributed to the initial page view. This is a limitation of the native web platform APIs on which CrUX is built, see [How SPA architectures affect Core Web Vitals](https://web.dev/vitals-spa-faq/) on web.dev for more information.

### Accelerated Mobile Pages (AMP)

Pages built with AMP are included in the CrUX dataset like any other web page as of the [June 2020 CrUX release](/docs/crux/release-notes/#202006). Pages served via the [AMP Cache](https://developers.google.com/amp/cache) and / or rendered in the [AMP Viewer](https://developers.google.com/search/docs/advanced/experience/about-amp#about-google-amp-viewer) are also captured, and attributed to the publisher's page URL.

## Tools

> The CrUX dataset is made available through a variety of tools maintained by Google. Each tool may access CrUX data slightly differently, resulting in varying levels of timeliness and metric support.

Reviewer 4: I'm biased but would be nice to add non Google tools :) useful for readers;

Reviewer 2: I do like supporting community tools but I don't think the official product docs are the right place to do that.

<cite><a href="https://web.dev/inp/">web.dev/inp/</a></cite>
{% endAside %}

Interaction to Next Paint (INP) was added to the CrUX dataset in [February 2022](/docs/crux/release-notes/#202202). This new metric captures the end-to-end latency of individual events and offers a more holistic picture of the overall responsiveness of a page throughout its lifetime.

#### Popularity {: #popularity-metric}

The [popularity rank](/blog/crux-rank-magnitude/) metric is a relative measure of site popularity within the CrUX dataset, measured by the total number of navigations on the origin. Rank is on a log10 scale (e.g. top 1k, 10k, 100k, 1M, etc.) with each rank excluding the previous (e.g. top 10k is actually 9k URLs, excluding top 1k). The upper limit is dynamic as the dataset grows.

> Popularity is provided as a guide for broad analysis, e.g. to determine performance by country for the top 1,000 origins. It should not be used to determine an origin's overall popularity on the web.

Author: I agree, I'll remove that statement.

Note that the specification defines four connection types, the majority of visits will likely be on connections faster than `3G` and thus be classified as `4G`:

<div class="responsive-table">
<table class="with-heading-tint width-full fixed-table">
<thead>
<tr>
<th>ECT</th>
<th>Minimum RTT</th>
<th>Maximum downlink</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>offline</td>
<td>N/A</td>
<td>N/A</td>
<td>The network is offline, only cached files can be served.</td>
</tr>
<tr>
<td>slow-2g</td>
<td>2000ms</td>
<td>50 Kbps</td>
<td>The network is suited for small transfers only such as text-only pages.</td>
</tr>
<tr>
<td>2g</td>
<td>1400ms</td>
<td>70 Kbps</td>
<td>The network is suited for transfers of small images.</td>
</tr>
<tr>
<td>3g</td>
<td>270ms</td>
<td>700 Kbps</td>
<td>The network is suited for transfers of large assets such as high resolution images, audio, and SD video.</td>
</tr>
<tr>
<td>4g</td>
<td>0ms</td>
<td>∞</td>
<td>The network is suited for HD video, real-time video, etc.</td>
</tr>
</tbody>
</table></div>

### Form Factor {: #form-factor-dimension }

The form factor dimension allows you to query against three separate form factors:

> - Mobile (`PHONE`)

Reviewer 4: Why in Uppercase?

Reviewer 2: That's how it appears in the CrUX API. Perhaps we don't need to use that convention outside of the API docs though. WDYT @simonhearne?

Reviewer 3: IMHO it should be in uppercase since it's in code format. Though there's a couple of other places we probably shoudl correct this. Will make comments.

Author: I don't think it's necessary to have the content in brackets at this point in the docs, I'm in favour of removing them and simplifying to :
- Phone
- Tablet
- Desktop

## Eligibility

At the core of the CrUX dataset are individual user experiences, which are aggregated into page-level and origin-level distributions. This section documents user eligibility and the requirements for pages and origins to be included in the dataset. All eligibility criteria must be satisfied in order for an experience to be included in page-level data available in PageSpeed Insights and the CrUX API: [User](#user-eligibility), [Origin](#origin-eligibility) and [Page](#page-eligibility). Experiences which meet the User and Origin criteria but not Page will be included in the origin-level data available in all CrUX data sources.

Pages and origins will be automatically included or removed from the dataset if their eligibility changes over time. There is not currently a way to manually submit pages or origins for inclusion.

> ### Publicly Discoverable {: #discoverability-eligibility }

Reviewer 2: Sentence case headings
```suggestion
### Publicly discoverable {: #discoverability-eligibility }
```

A page must be publicly discoverable to be considered for inclusion in the CrUX dataset.

A page is determined to be publicly discoverable using the same [indexability](https://developers.google.com/search/docs/advanced/crawling/block-indexing) and [crawlability](https://developers.google.com/search/docs/advanced/robots/intro) criteria as search engines.

Any page will **not** meet the discoverability requirement if **any** of the following conditions are met, including root pages for the origin dataset:

- The page is served with an HTTP [status code](https://developer.mozilla.org/docs/Web/HTTP/Status) other than `200` (after redirects).
- The page is served with an HTTP `X-Robots-Tag: noindex` [header](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag#xrobotstag-implementation).
- The document includes a `<meta name="robots" content="noindex">` [meta tag](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag).
- The URL is disallowed by [robots.txt](https://developers.google.com/search/docs/advanced/robots/intro).

> ### Sufficiently Popular {: #popularity-eligibility }

Reviewer 2: ```suggestion
### Sufficiently popular {: #popularity-eligibility }
```
